state/txindex/kv/kv.go

```diff
@@ -923,3 +1003,24 @@ func startKey(fields ...any) []byte {
 	}
 	return b.Bytes()
 }
+
+func validatePage(pagePtr *int, perPage, totalCount int) (int, error) {
+	if perPage < 1 {
+		panic(fmt.Sprintf("zero or negative perPage: %d", perPage))
```

Comment:
shouldn't we error here?

Reply:
I am actually just copying the logic from here https://github.com/cometbft/cometbft/blob/ca4090ea8147496d945b06bd11f907cc305666cf/rpc/core/env.go#L98-L100

Since we cant make the function public and import due to cyclic deps, I had to duplicate here. 

We called this method previously (one level higher in Search), and I wanted to keep the same expectations as there was previously, just essentially switching the order of events.

After seeing this lmk if you would still like me to change to error and I will do so!

Reply:
better change to error

Reply:
Done here https://github.com/cometbft/cometbft/pull/2855/commits/0e7ac0a44d88b0586450f63114dc82153a2701b8